python/ray/train/gbdt_trainer.py

```diff
@@ -193,6 +223,21 @@ def training_loop(self) -> None:
             **config,
         )
 
+        checkpoint_at_end = self.run_config.checkpoint_config.checkpoint_at_end
+        if checkpoint_at_end is None:
+            checkpoint_at_end = True
```

Comment:
providing a default in customer class seems strange.
should we just give a True default in CheckpointConfig?

Reply:
In this case we can't. We have different default values for checkpoint_at_end depending on the trainable. If we set it to True per default, it will be incompatible with legacy trainables such as regular function trainables and raise an error downstream in tune.run. If we set it to False per default, we won't save checkpoints for most trainers and can't use their models in downstream processing.

I'm working on a FunctionTrainer today that will replace running legacy trainable functions. This will enable us to ignore the legacy function trainables here and default to True. Does that sound good?

Reply:
Note that another reason why we can't to it at the moment is that if people specifically pass `checkpoint_at_end=True` to use with their function trainables, we don't want to silently set it to False, but we could do this once we use a FunctionTrainer (though it's not ideal as users may still pass True and wonder why they don't see any saved checkpoints)

Reply:
I see. appreciate the detailed explanation.
navigating through a lot of legacy stuff is hard ...